nixos/modules/tasks/lvm.nix

```diff
@@ -47,13 +57,15 @@ in {
       boot.initrd = {
         kernelModules = [ "dm-snapshot" "dm-thin-pool" ];
 
-        extraUtilsCommands = ''
+        systemd.initrdBin = lib.mkIf config.boot.initrd.services.lvm.enable [ pkgs.thin-provisioning-tools ];
+
+        extraUtilsCommands = mkIf (!config.boot.initrd.systemd.enable) ''
           for BIN in ${pkgs.thin-provisioning-tools}/bin/*; do
             copy_bin_and_libs $BIN
           done
         '';
 
-        extraUtilsCommandsTest = ''
+        extraUtilsCommandsTest = mkIf (!config.boot.initrd.systemd.enable) ''
```

Comment:
Would it be possible to just ignore these settings at a global level when `boot.initrd.systemd.enable` is set?

That way we don't need this clause everywhere.

Reply:
Yeah they are just ignored.
But my idea was to get rid of their uses and assert that the options are not set to make sure people don't set these options in their local configs and being surprised they don't do anything anymore.

Reply:
Good idea!